src/util/mediatypes.js

```
@@ -0,0 +1,117 @@
+/* This Source Code Form is subject to the terms of the MIT license
+ * If a copy of the MIT license was not distributed with this file, you can
+ * obtain one at https://raw.github.com/mozilla/butter/master/LICENSE */
+
+"use strict";
+
+define( [ "util/xhr", "util/uri" ],
+  function( XHR, URI ) {
+
+  var REGEX_MAP = {
+    youtube: /(?:https?:\/\/www\.|https?:\/\/|www\.|\.|^)youtu/,
+    vimeo: /https?:\/\/(www\.)?vimeo.com\/(\d+)($|\/)/,
+    soundcloud: /(?:https?:\/\/www\.|https?:\/\/|www\.|\.|^)(soundcloud)/
+  };
+
+  var EXAMPLES = {
+    youtube: "http://www.youtube.com/watch?v=CLg2JbELs9o",
+    html5: "http://popcorn.webmadecontent.org/videos/getstarted.mp4"
+  };
+
+  return {
+    checkUrl: function( url ) {
+      for ( var type in REGEX_MAP ) {
+        if ( REGEX_MAP.hasOwnProperty( type ) ) {
+          if ( REGEX_MAP[ type ].test( url ) ) {
+            return type;
+          }
+        }
+      }
+      return "html5";
+    },
+    getMetaData: function( baseUrl, callback, type ) {
+      var data = {},
+          id,
+          parsedUri,
+          splitUriDirectory,
+          xhrURL,
+          testEl;
+
+      if ( baseUrl === "test" ) {
+        baseUrl = EXAMPLES.youtube;
+      }
+
+      type = data.type = type || this.checkUrl( baseUrl );
+      callback = callback || function(){};
+
+      if ( type === "youtube" ) {
+        parsedUri = URI.parse( baseUrl );
+        id = parsedUri.queryKey.v || parsedUri.directory.replace( "/", "" );
+        if ( !id ) {
+          return;
```

So the type was "youtube" but you failed to get an id and you're just going to silently return?  This is going to hide a bug if we get the parsing of their URLs wrong, no?  Should we not do something in the editor that says, "This YouTube URL doesn't seem to be valid, please check..."